dropshot/src/handler.rs

```diff
@@ -888,6 +894,78 @@ where
     }
 }
 
+/*
+ * UntypedBody: body extractor for a plain array of bytes of a body.
+ */
+
+/**
+ * `UntypedBody` is an extractor for reading in the contents of the HTTP request
+ * body and making the raw bytes directly available to the consumer.
+ */
+pub struct UntypedBody {
+    content: Bytes,
+}
+
+impl UntypedBody {
+    /**
+     * Returns a byte slice of the underlying body content.
+     */
+    /*
+     * TODO drop this in favor of Deref?  + Display and Debug for convenience?
+     */
+    pub fn as_bytes(&self) -> &[u8] {
+        &self.content
+    }
+
+    /**
+     * Convenience wrapper to convert the body to a UTF-8 string slice,
+     * returning a 400-level error if the body is not valid UTF-8.
+     */
+    pub fn as_str(&self) -> Result<&str, HttpError> {
+        std::str::from_utf8(self.as_bytes()).map_err(|e| {
+            HttpError::for_bad_request(
+                None,
+                format!("failed to parse body as UTF-8 string: {}", e),
+            )
+        })
+    }
+}
+
+#[async_trait]
+impl Extractor for UntypedBody {
+    async fn from_request(
+        rqctx: Arc<RequestContext>,
+    ) -> Result<UntypedBody, HttpError> {
+        let server = &rqctx.server;
+        let mut request = rqctx.request.lock().await;
+        let body_bytes = http_read_body(
+            request.body_mut(),
+            server.config.request_body_max_bytes,
+        )
+        .await?;
+        Ok(UntypedBody {
+            content: body_bytes,
+        })
+    }
+
+    fn metadata() -> Vec<ApiEndpointParameter> {
+        let schema = SchemaObject {
+            instance_type: Some(InstanceType::String.into()),
```

Comment:
well that's a lot simpler than what I did here: https://github.com/oxidecomputer/oxide-api-prototype/blob/master/src/api_model.rs#L182

love it

Reply:
Yeah, I saw this in one of the implementations in `schemars`.  I normally avoid `into()`.